TrustModel.md

```diff
@@ -0,0 +1,426 @@
+memory inside the environment from access in the clear by any actor outside the environment. 
+ANd core to the trust model is where within a cloud native 
+stack to 
+put the Trusted Execution 
+boundary when looking to deliver confidential containers.
+
+Three main boundaries for the Trusted Execution Boundary are:
+- Node
+- Pod
+- Container/Container Processes
+
+<*Add Diagram*>
+
+
+Each boundary changes requirements on what components must exist inside the TEE, the 
+form the image to run inside the TEE takes and ultimately on the trust model.
+
+### Node boundary
+This uses a virtual machine with hardware TEE capability to run a K8s node.
+The kubelet is inside the TEE boundary and therefore communication between api-server and 
+kubelet would cross the boundary.
+The 
+
+### Pod Boundary
+This uses a micro virtual machine (microVM) with hardware TEE capability.
+The microVM contains an agent 
+
+### Container/Container Process boundary
+This uses an enclave to run a LibOS (Intel SGX Enclave)
+
+
+In all cases the application code interacts with data within a hardware TEE
+
+A brief comparison:
+
+| | node | pod  | container/process |
+| | vm-cc | microvm-cc/kata-cc\*1 | enclave-cc |
+| :-: | :-: | :-: | :-: |
+| TCB footprint | Large | Medium | Small |
+| Location of trust boundary\*2 | between K8s API server and kubelet | agent API\*3 | ECALL and OCALL |
+| Threat Surface | High | Medium | Low |
+| Deployment complexity | High | Low | Medium |
+| Changes to workload Orchestration Experience | Low | Medium | High |
+
+Note:
+1. This doc may use kata-cc equivalent to microvm-cc in order to refer to a kata-specific component.
+2. Here refers to the trust boundary of control plane.
+3. A component used in kata and running in POD.
+
+
+Efforts so far have focused on the pod boundary using Kata containers as the technology to run a 
+pod within the TEE boundary. We expect this document to align the pod and container approaches 
+over time, but we believe that the node approach cannot deliver the required trust for 
+confidential containers without significant changes throughout the stack which are currently 
+beyond the scope of our efforts.
+
+## Conclusion
+
+The trust model determines the location and direction of the trust boundary. Trust boundary
+describes a location where program data or execution changes its level of "trust," or where two
+principals with different capabilities exchange data or commands[3]. The trust boundary of
+confidential container is also the trust boundary of TEE. Generally, the TEE side at the trust
+boundary should be hardened to prevent the violation of the trust boundary.
+
+In general, the trust model adopted by any confidential container based solutions can be binarized
+into:
+1) Partially trust the platform owner, operator and privileged components controlled by host
+2) Completely distrust the platform owner, operator and privileged components controlled by host
```

Comment:
I think we're almost certainly aiming for the second one, right? Would we be satisfied at all with partially trusting the CSP? 

One term that I really like is `deprivileging orchestration` which I think describes most of the work that we have done on control-plane components so far; reworking the control plane so that it can handle orchestration without breaking confidentiality. I think the split between orchestration and confidentiality is crucial. Maybe we can find a way to work it into this document.

Reply:
I agree we are more focused on the second, I am now working to split this across particular aspects of the trust model now. So given a way of thinking about the trust model (persona v TEE boundary v lifecyle) we can highlight what is our big concern trust wise.

I do like your phrase deprivileging orchestration too :-) I have worked it into the Conclusion for the Trust Model Personas.

Changes made in PR -> https://github.com/confidential-containers/documentation/pull/24
We can continue Trust Model Persona discussion in this new specific PR